source/secure-element/README.md

```diff
@@ -0,0 +1,160 @@
+`secure-element-pin`: User PIN for logging into PKCS#11 token.
+
+**Optional Parameter:**
+
+`secure-element-key-label`: Label of private key on the PKCS#11 token (optional).
+
+`secure-element-slot-id`: Numeric Slot ID containing PKCS#11 token to use (optional).
+
+`secure-element-token-label`: Label of the PKCS#11 token to use (optional).
+
+#### Configuring the Secure Element feature via the command line
+```
+$ ./aws-iot-device-client --enable-secure-element [true|false] --pkcs11-lib [your/path/to/pkcs#11/library] --secure-element-pin [User PIN of PKCS#11 token] --secure-element-key-label [key-label] --secure-element-slot-id [token-slot-id] --secure-element-token-label [token-label]
+```
+#### Configuring the Secure Element feature via the JSON configuration file
+```
+{
+    ...
+    "secure-element": {
+		"enabled": false,
+		"pkcs11-lib": "<replace_with_pkcs11_lib_path>",
+		"secure-element-pin": "<replace_with_secure_element_pin>",
+		"secure-element-key-label": "<replace_with_secure_element_key_label>",
+		"secure-element-slot-id": replace_with_secure_element_slot_id_integer,
+		"secure-element-token-label": "<replace_with_secure_element_token_label>"
+	}
+    ...
+}
+```
+
+**Note:** If Secure Element feature is disabled, Device Client will look for private key on the file system.
+
+
+## Building And Starting Device Client with Secure Element Feature
+
+The following example shows how user can build and start the Device Client with Secure Element feature enabled. In this example, we will use SoftHSM2 for storing private key. 
+
+### Building Device Client
+
+The following commands should work for most users when you plan to run the AWS IoT Device Client on the same machine that you're performing the compilation on:
+
+```
+# Building
+git clone https://github.com/awslabs/aws-iot-device-client
+cd aws-iot-device-client
+mkdir build
+cd build
+cmake -DEXCLUDE-SECURE-ELEMENT="OFF" ../
+cmake --build . --target aws-iot-device-client
+
+# Setup
+cd ../
+./setup.sh # At this point you'll need to respond to prompts for information, including paths to your thing certs
+
+```
+
+**Note:** The user can use `EXCLUDE-SECURE-ELEMENT` cmake option for including or excluding Secure Element feature from binary build. Default value of this flag is set to `ON`.
+
+**Note:** If the user decides not to use `setup.sh` script, they can manually update the JSON config file or use CLI commands to set the secure element config parameters as shown above.
+
+### Configure SoftHSM2 and start Device Client 
+
+In this example, we will use SoftHSM2 for storing private key. 
+
+1. Create an IoT Thing with a certificate and key if you haven't already.
+2. Convert the private key into PKCS#8 format:
+
+   ```openssl pkcs8 -topk8 -in <private.pem.key> -out <private.p8.key> -nocrypt```
+3. Install SoftHSM2:
+
+    ```apt install softhsm```
+4. Check that it's working:
+
+   ```softhsm2-util --show-slots```
+
+   If this spits out an error message, create a config file:
```

Comment:
Please mention: You will also need to export the location of the config file using the `SOFTHSM2_CONF`. Example below:

```
export SOFTHSM2_CONF=/path/to/config/softhsm2.conf
```